deny.toml

```diff
@@ -59,10 +59,12 @@ skip = [
     # Since `syn` is used by proc-macros (executed at compile time), duplicate
     # versions won't have an impact on the final binary size.
     { name = "syn" },
-]
-skip-tree = [
-    # Hasn't seen a new release since 2017. Pulls in an older version of nom.
-    { name = "procinfo" },
+    # right now we have a mix of versions of this crate in the ecosystem
+    # procfs uses 0.36.14, tempfile uses 0.37.4
+    { name = "rustix" },
```

Comment:
it might be worth opening a PR upstream to update `procfs`' dependency on `rustix`? then, we would be able to remove this.

Reply:
So the latest version of rustix is 0.38.0
If we are planning to upgrade then should we upgrading the rustix version for both the dependencies ? Or shall we ignore it for now ?

Reply:
Ah, sorry, I didn't see this. It would be ideal to get both dependencies on the latest `rustix`. That isn't a blocker for merging this PR, but it could be nice to kick off.

Reply:
Yes, it would be great. But help me understand something, I just cannot see where we are using the `tempfile` crate in our code base (this is another crate that uses `rustix` and the reason why we need to add it into deny.toml). I might be making a mistake here but in case we are not using it I can go ahead and remove it.

Reply:
It looks like `tempfile` is a transitive dependency of `prost-build`, which is used for compiling protobufs. This is a build-time dependency, so it's not actually used by the proxy itself. Instead, it's used when building the proxy, to generate Rust code from protocol buffer files.

We can determine where a transitive dependency comes from by using [`cargo tree`](https://doc.rust-lang.org/cargo/commands/cargo-tree.html). If we pass the `--invert` flag to `cargo tree`, we get a tree whose root is the named crate and whose leaves are the crates that depend on that crate. For `tempfile`, it looks like this:
```console
$ cargo tree --package tempfile --invert
tempfile v3.5.0
└── prost-build v0.11.8
    └── tonic-build v0.8.4
        └── tools v0.1.0 (/home/eliza/Code/linkerd2-proxy/tools)
        [dev-dependencies]
        └── opencensus-proto v0.1.0 (/home/eliza/Code/linkerd2-proxy/opencensus-proto)
            └── linkerd-opencensus v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/opencensus)
                ├── linkerd-app v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app)
                │   ├── linkerd-app-integration v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/integration)
                │   └── linkerd2-proxy v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd2-proxy)
                └── linkerd-app-core v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/core)
                    ├── linkerd-app v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app) (*)
                    ├── linkerd-app-admin v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/admin)
                    │   └── linkerd-app v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app) (*)
                    │   [dev-dependencies]
                    │   └── linkerd-app-integration v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/integration)
                    ├── linkerd-app-gateway v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/gateway)
                    │   └── linkerd-app v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app) (*)
                    ├── linkerd-app-inbound v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/inbound)
                    │   ├── linkerd-app v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app) (*)
                    │   ├── linkerd-app-admin v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/admin) (*)
                    │   └── linkerd-app-gateway v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/gateway) (*)
                    │   [dev-dependencies]
                    │   └── linkerd-app-gateway v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/gateway) (*)
                    ├── linkerd-app-integration v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/integration)
                    ├── linkerd-app-outbound v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/outbound)
                    │   ├── linkerd-app v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app) (*)
                    │   └── linkerd-app-gateway v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/gateway) (*)
                    │   [dev-dependencies]
                    │   └── linkerd-app-gateway v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/gateway) (*)
                    └── linkerd-app-test v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/test)
                        ├── linkerd-app-inbound v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/inbound) (*)
                        ├── linkerd-app-integration v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/integration)
                        └── linkerd-app-outbound v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/outbound) (*)
                        [dev-dependencies]
                        ├── linkerd-app-gateway v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/gateway) (*)
                        ├── linkerd-app-inbound v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/inbound) (*)
                        └── linkerd-app-outbound v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/outbound) (*)
    [dev-dependencies]
    └── linkerd-transport-header v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/transport-header)
        └── linkerd-app-core v0.1.0 (/home/eliza/Code/linkerd2-proxy/linkerd/app/core) (*)
```